src/test/kotlin/com/example/passmanager/repositories/PassOwnerRepositoryImplTest.kt

```diff
@@ -31,70 +30,70 @@ class PassOwnerRepositoryImplTest {
         // WHEN
         val insertedPassOwner = mongoTemplate.insert(getOwnerWithUniqueFields())
-        val insertedPassOwnerId = insertedPassOwner.id
 
         // THEN
-        assertThat(mongoTemplate.findById<MongoPassOwner>(insertedPassOwnerId!!)).isEqualTo(insertedPassOwner)
+        insertedPassOwner
+            .mapNotNull { it.id }
+            .test()
+            .assertNext { ownerId ->
+                mongoTemplate.findById<MongoPassOwner>(ownerId!!)
+                    .mapNotNull { it.id }
+                    .doOnNext { id -> assertThat(ownerId).isEqualTo(id) }
+                    .subscribe()
+            }
+            .verifyComplete()
     }
 
     @Test
     fun `saving pass owner in collection show update existing pass owner by id`() {
         // GIVEN
         val insertedPassOwner = mongoTemplate.insert(getOwnerWithUniqueFields())
         val changedFirstName = "Changed first name"
-        val updatedPassOwner = insertedPassOwner.copy(firstName = changedFirstName)
+        val updatedPassOwner = insertedPassOwner.map { it.copy(firstName = changedFirstName) }
 
         // WHEN
-        val saved = passOwnerRepository.save(updatedPassOwner)
+        val saved = updatedPassOwner.flatMap { passOwnerRepository.save(it) }
 
         // WHEN
-        assertThat(saved.firstName).isEqualTo(changedFirstName)
+        saved.test()
+            .assertNext { assertThat(it.firstName).isEqualTo(changedFirstName) }
+            .verifyComplete()
     }
 
     @Test
     fun `deleting pass owner by id should delete pass owner from collection`() {
         // GIVEN
-        val insertedPassOwner = mongoTemplate.insert(getOwnerWithUniqueFields())
-        val insertedPassOwnerId = insertedPassOwner.id
-
-        // WHEN
-        passOwnerRepository.deleteById(insertedPassOwnerId.toString())
-
-        // THEN
-        assertFalse("pass owner must not exist in collection after deletion") {
-            mongoTemplate.exists<MongoPassOwner>(query(where("_id").isEqualTo(insertedPassOwnerId)))
-        }
-    }
-
-    @Test
-    fun `optimistic lock handling while save() should throw exception if version was changed by another thread()`() {
-        // GIVEN
-        val insertedPassOwner = passOwnerRepository.insert(passOwnerToCreate)
-        val firstNames = listOf("Firstname1", "Firstname2")
-        assertEquals(1, insertedPassOwner.version)
+        // Cache the result to avoid re-triggering insert
+        val insertedPassOwner = mongoTemplate.insert(getOwnerWithUniqueFields()).cache()
 
         // WHEN
-        val tasks = firstNames.map {
-            Runnable { passOwnerRepository.save(insertedPassOwner.copy(firstName = it)) }
-        }
-        val optimisticLocks = getOptimisticLocksAmount(tasks)
+        insertedPassOwner
+            .mapNotNull { it.id.toString() }
+            .flatMap { ownerId -> passOwnerRepository.deleteById(ownerId) }
+            .subscribe()
```

Comment:
Why subscribe ?
If you want to verify that delete method was executed, you can use 
```kotlin
.test()
.verifyComplete()
```

Reply:
I want to verify entity was deleted from collection, not only deleteById method execution